src/Kubernetes/KubernetesApiExtensions.cs

```diff
@@ -503,6 +503,31 @@ public static V1Job WithBillingId(this V1Job job, string billingId)
             });
         }
 
+        /// <summary>
+        /// Adds a policy failure action and exit codes to the job.
+        /// </summary>
+        /// <param name="job">The job object to modify.</param>
+        /// <param name="actionExitCodeId">A list with all action names and corresponding exit code number.</param>
+        /// <returns>The Kubernetes Job object with added pod failure policy rules.</returns>>
+        public static V1Job WithPodPolicyFailureExitCodes(this V1Job job, List<(string action, int exitCode)> actionExitCodeId)
+        {
+            var groupedActions = actionExitCodeId.GroupBy(aec => aec.action);
+
+            job.Spec ??= new V1JobSpec();
+            job.Spec.PodFailurePolicy ??= new V1PodFailurePolicy();
+
+            job.Spec.PodFailurePolicy.Rules = groupedActions.Select(actionName => new V1PodFailurePolicyRule
```

Comment:
would be easier with a `Dictionary` as mentioned before. Plus you must check that `Rules` exist, and if not, your logic applies, if yes, you need to combine them with what's added, otherwise this method will always replace the rules - unless this is the intended behaviour?

Reply:
With dictionary is much easier, replacing the rules was not the intended behavior so your suggestion proved to be very good 👍 
I've updated it now in [3822912](https://github.com/SneaksAndData/esd-services-sdk/pull/75/commits/38229123666d315f956b5f23e8cf98a9a1a445d5)
Thank you for the input 🙂